src/linux/sock_ctrl_msg.rs

```diff
@@ -219,29 +220,42 @@ fn raw_recvmsg(fd: RawFd, iovecs: &mut [iovec], in_fds: &mut [RawFd]) -> Result<
         return Ok((0, 0));
     }
 
+    // Reference to a memory area with a CmsgBuffer, which contains a `cmsghdr` struct followed
```

Comment:
Seems we should check MSG_TRUNC  and MSG_CTRUNC  in msghdr.msg_flags for safety.

Reply:
Thanks for suggesting this. Indeed, checking for `MSG_CTRUNC` (`MSG_TRUNC` is not relevant for the scope of the PR) brought more light to how to assure the safety of this function. I've found out that there are two cases we need to be aware about (added unit tests for each):

- `raw_recvmsg` allocates enough data for receiving the ancillary data (within the `CmesgBuffer`) - it is computed based on the buffer received by the function from the outer scope, but it is also aligned, adding extra capacity which in these case accounts for few extra fds that can not be stored in the buffer from the outer scope. For this case I've added code paths to close the extra `fds`.
- `raw_recvmsg` does not allocate enough data for receiving the ancillary data and implied, the buffer from the outer scope as well, so the inner call to `recvmsg` sets the mshdr flag `MSG_CTRUNC` and drops any extra ancillary data. For this case we do not need to handle any extra data.

@jiangliu , please let me know if all of the above make sense to you.

Reply:
I'm hesitating about what to return in case of receiving unexpected data/fds. It's a sign of incorrect communication protocol, so it's a little risky to silently drop the unexpected data/fds. How about returning an error here?

Reply:
Initially I've thought providing a best effort implementation, but as you pointed out, its a sign of incorrect communication protocol, and it would be safer to exclude it from the functionality altogether. With this picture in mind and for a clear expectation for the end users, I think it would be ideal to drop all the received fds from the control data - the ones that can be held in the buffer and the extra ones as well - before returning the error. WDYT?

Reply:
exactly:)